### ISSUE TITLE
Incorrect error in EPUB 3 when a col tag is not a child of a colgroup

### ISSUE THREAD

**Issue by aparod** [IDPF/epubcheck#628]
Consider the following table, which has col tags nested directly under the table tag:

```html
<table>
    <col style="width: 10%" />
    <col style="width: 90%" />
    <tr>
        <td>Cell 1</td>
        <td>Cell 2</td>
    </tr>
</table>
```

This generates no error when part of an EPUB 2, as it is valid XHTML 1.1.  Here's the relevant part of [the DTD](http://www.w3.org/MarkUp/SCHEMA/xhtml-table-1.xsd), where the col tag is explicitly allowed as a child of the table tag:

```xml
<xs:group name="xhtml.table.content">
    <xs:sequence>
        <xs:element name="caption" type="xhtml.caption.type" minOccurs="0"/>
        <xs:choice>
            <xs:element name="col" type="xhtml.col.type" minOccurs="0" maxOccurs="unbounded"/>
            <xs:element name="colgroup" type="xhtml.colgroup.type" minOccurs="0" maxOccurs="unbounded"/>
        </xs:choice>

        ....

    </xs:sequence>
</xs:group>
```

However, the above table, when part of an EPUB 3, generates the following error:

    element "col" not allowed here; expected the element end-tag or element "caption", "colgroup", "tbody", "tfoot", "thead" or "tr"

The HTML 5 specs indicate that the colgroup's start and end tags are optional and can be inferred, just as was the case in XHTML 1.1. See [the colgroup tag on MDN](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/colgroup).

It seems like [EpubCheck's HTML 5 schema](https://github.com/IDPF/epubcheck/blob/master/src/main/resources/com/adobe/epubcheck/schema/30/mod/html5/html5-grouping-30.rnc#L72) should be updated to allow for this.  Here's the current version:

```
table.content = (html5.caption?, html5.colgroup*, html5.thead?, 
    ((html5.tfoot, ( html5.tbody* | html5.tr+ ))|(( html5.tbody* | html5.tr+ ), html5.tfoot?)))
```

and my proposed solution:

```
table.content = (html5.caption?, (html5.colgroup* | html5.col*), html5.thead?, 
    ((html5.tfoot, ( html5.tbody* | html5.tr+ ))|(( html5.tbody* | html5.tr+ ), html5.tfoot?)))
```

I'll submit a pull request.

**Comment by mattgarrish** [comment]
You're reading the whatwg colgroup definition, which differs slightly from the w3c that epub references (the w3c spec requires an opening tag in all cases).

But regardless of which you choose, the tag omission is only valid for the text/html serialization that epub doesn't support, so epubcheck is correct.

**Comment by mattgarrish** [comment]
It looks like w3c harmonizes again in 5.1, but it's still only for text/html.

**Comment by rdeltour** [comment]
Thanks Matt. Closing this as invalid.
